File: week_1/day-3-two-pointer/two_pointers_warmup/two_sum_unsorted.js
const nums = [3, 1, 4, 2];
const target = 6;

function twoSumUnsorted(nums, target) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (map.has(target - nums[i])) {
            return [map.get(target - nums[i]), i];
        }
        map.set(nums[i], i);
    }
}

console.log(twoSumUnsorted(nums, target)); // [2, 3]

File: week_1/day-3-two-pointer/two_pointers_warmup/two_sum_unsorted.py
nums = [3, 1, 4, 2]
target = 6

def two_sum_unsorted(nums, target):
    seen = {}
    for i, num in enumerate(nums):
        if target - num in seen:
            return seen[target - num], i
        seen[num] = i

print(two_sum_unsorted(nums, target))  # Output: (2, 3)
